### PR TITLE
Set secure request attribute correctly in blaze-server

### LINKED ISSUE #1104: BlazeServer hardcodes Request.Connection(secure = true)
After the following commit  https://github.com/http4s/http4s/commit/104d1f59a59263f6f6c73890fb7b72e6331f62a3 
Request attributes always contain secure = true

```scala
val requestAttributes =
  (conn.local, conn.remote) match {
    case (l: InetSocketAddress, r: InetSocketAddress) =>
      AttributeMap(AttributeEntry(Request.Keys.ConnectionInfo, Request.Connection(l,r, true)))
    case _ =>
      AttributeMap.empty
  }
```

### PR DESCRIPTION
**Pull request by rossabaker**
Fixes #1104 

/cc @Unisay

### CODE REVIEW

**Comment by Unisay** [comment]
:+1:

File: blaze-server/src/main/scala/org/http4s/server/blaze/BlazeServer.scala [hunk 1]
```diff
@@ -158,14 +162,14 @@ class BlazeBuilder(
 
           var lb = LeafBuilder(
             if (isHttp2Enabled) http2Stage(engine)
-            else http1Stage
+            else http1Stage(true)
```
Comment (Unisay) [inline blaze-server/src/main/scala/org/http4s/server/blaze/BlazeServer.scala]: I believe that such argument should use a named-parameter syntax: `http1Stage(secure = true)`

File: blaze-server/src/main/scala/org/http4s/server/blaze/BlazeServer.scala [hunk 2]
```diff
@@ -158,14 +162,14 @@ class BlazeBuilder(
...
           )
           lb = prependIdleTimeout(lb)
           lb.prepend(new SSLStage(engine))
 
         case None =>
           if (isHttp2Enabled) logger.warn("HTTP/2 support requires TLS.")
-          var lb = LeafBuilder(http1Stage)
+          var lb = LeafBuilder(http1Stage(false))
```
Comment (Unisay) [inline blaze-server/src/main/scala/org/http4s/server/blaze/BlazeServer.scala]: `http1Stage(secure = false)` looks better

File: blaze-server/src/main/scala/org/http4s/server/blaze/BlazeServer.scala [hunk 3]
```diff
@@ -158,14 +162,14 @@ class BlazeBuilder(
...
           var lb = LeafBuilder(
             if (isHttp2Enabled) http2Stage(engine)
-            else http1Stage
+            else http1Stage(secure = true)
           )
           lb = prependIdleTimeout(lb)
           lb.prepend(new SSLStage(engine))
 
         case None =>
           if (isHttp2Enabled) logger.warn("HTTP/2 support requires TLS.")
```
Comment (bfritz) [inline blaze-server/src/main/scala/org/http4s/server/blaze/BlazeServer.scala]: Off-topic for this PR, but should that warning say something about falling back to HTTP1?

Comment (rossabaker) [inline blaze-server/src/main/scala/org/http4s/server/blaze/BlazeServer.scala]: Yes.

**Pull request merged**